frontend/src/CodalabApp.js

```diff
@@ -101,7 +101,7 @@ function CodalabApp() {
                                 render={(props) => <Login {...props} auth={fakeAuth} />}
                             />
                             <PrivateRoute path='/account/profile' component={UserInfo} />
-                            <Route path='/worksheets/:uuid' component={Worksheet} />
+                            <Route path='/worksheets/:uuid/:mode?' component={Worksheet} />
```

Comment:
I think it would be best to remove the URL change when expanding / shrinking, given that this is a global setting that is set in localStorage.